src/merkle/tiered_smt/tests.rs

```diff
@@ -0,0 +1,189 @@
+use super::*;
+use crate::{utils::collections::BTreeSet, ONE, WORD_SIZE};
+use core::ops::{Deref, DerefMut};
+use proptest::prelude::*;
+use proptest::sample::size_range;
+
+#[test]
+fn append_to_bottom_level_is_consistent() {
+    let mut test = TieredTestEngine::default();
+
+    // insert a leaf into the empty tree. should be allocated to the first tier.
+    let raw_a = 0b_01101001_01101100_00011111_11111111_10010110_10010011_11100000_00000000_u64;
+    let key_a = [Felt::new(raw_a); WORD_SIZE];
+    test.insert(key_a, Word::default())
+        .expect_path_with_depth(key_a, 16);
+
+    // insert another leaf and diverge on the first bit, causing both `a` and `b` to be on depth
+    // `16. should produce an empty path until the last element.
+    let raw_b = 0b_11101001_01101100_00011111_11111111_10010110_10010011_11100000_00000000_u64;
+    let key_b = [Felt::new(raw_b); WORD_SIZE];
+    test.insert(key_b, Word::default())
+        .expect_path_with_depth(key_a, 16)
+        .expect_path_with_depth(key_b, 16);
+
+    // this new leaf will have the same path of `a` until the first bit of the second tier, where
+    // they diverge. both `a` and `c` should be on the depth `32` after this mutation.
+    let raw_c = 0b_01101001_01101100_10011111_11111111_10010110_10010011_11100000_00000000_u64;
+    let key_c = [Felt::new(raw_c); WORD_SIZE];
+    // expect the path of `a` to be mutated as well.
+    test.insert(key_c, Word::default())
+        .expect_path_with_depth(key_a, 32)
+        .expect_path_with_depth(key_b, 16)
+        .expect_path_with_depth(key_c, 32);
+
+    // this new leaf will have the same path of `c` until the first bit of the third tier, where
+    // they diverge. both `c` and `d` should be on the depth `48` after this mutation.
+    let raw_d = 0b_01101001_01101100_10011111_11111111_00010110_10010011_11100000_00000000_u64;
+    let key_d = [Felt::new(raw_d); WORD_SIZE];
+    // expect the path of `c` to be mutated as well.
+    test.insert(key_d, Word::default())
+        .expect_path_with_depth(key_a, 32)
+        .expect_path_with_depth(key_b, 16)
+        .expect_path_with_depth(key_c, 48)
+        .expect_path_with_depth(key_d, 48);
+
+    // this new leaf will have the same path of `d` ultil the first bit of the fourth tier, where
+    // they will diverge. both `d` and `e` should be on the depth `64` after this mutation.
+    let raw_e = 0b_01101001_01101100_10011111_11111111_00010110_10010011_01100000_00000000_u64;
+    let key_e = [Felt::new(raw_e); WORD_SIZE];
+    // expect the path of `d` to be mutated as well.
+    test.insert(key_e, Word::default())
+        .expect_path_with_depth(key_a, 32)
+        .expect_path_with_depth(key_b, 16)
+        .expect_path_with_depth(key_c, 48)
+        .expect_path_with_depth(key_d, 64)
+        .expect_path_with_depth(key_e, 64);
```

Comment:
After this, I'd also add a check that `get_bottom_leaves()` retrieves expected vectors for both `d` and `e`.

Reply:
Done; added a `expect_bottom_leaves` test function